src/qt/walletmodel.cpp

```diff
@@ -215,11 +215,6 @@ WalletModel::SendCoinsReturn WalletModel::prepareTransaction(WalletModelTransact
         return OK;
     }
 
-    if(isAnonymizeOnlyUnlocked())
-    {
-        return AnonymizeOnlyUnlocked;
-    }
-
```

Comment:
Shouldn't this function be checking the IsLocked status ?  It looks like it accesses the wallet without checking.